File: bicitaxi/lib/features/rides/models/ride_status.dart
/// Status of a ride throughout its lifecycle.
enum RideStatus {
  /// Client has requested a ride.
  requested,

  /// System is searching for an available driver.
  searchingDriver,

  /// A driver has been assigned to the ride.
  driverAssigned,

  /// Driver is en route to pickup location.
  driverArriving,

  /// Ride is in progress (passenger on board).
  inProgress,

  /// Ride has been completed successfully.
  completed,

  /// Ride was cancelled.
  cancelled,
}

/// Extension to provide display names for ride statuses.
extension RideStatusExtension on RideStatus {
  String get displayName {
    switch (this) {
      case RideStatus.requested:
        return 'Solicitado';
      case RideStatus.searchingDriver:
        return 'Buscando conductor';
      case RideStatus.driverAssigned:
        return 'Conductor asignado';
      case RideStatus.driverArriving:
        return 'Conductor en camino';
      case RideStatus.inProgress:
        return 'En progreso';
      case RideStatus.completed:
        return 'Completado';
      case RideStatus.cancelled:
        return 'Cancelado';
    }
  }

  /// Returns the next status in the workflow (for simulation).
  RideStatus? get nextStatus {
    switch (this) {
      case RideStatus.requested:
        return RideStatus.searchingDriver;
      case RideStatus.searchingDriver:
        return RideStatus.driverAssigned;
      case RideStatus.driverAssigned:
        return RideStatus.driverArriving;
      case RideStatus.driverArriving:
        return RideStatus.inProgress;
      case RideStatus.inProgress:
        return RideStatus.completed;
      case RideStatus.completed:
      case RideStatus.cancelled:
        return null;
    }
  }

  /// Whether the ride is still active (not terminal).
  bool get isActive {
    return this != RideStatus.completed && this != RideStatus.cancelled;
  }
}

File: ios/bicitaxi/bicitaxi/RideStatus.swift
//
//  RideStatus.swift
//  bicitaxi
//
//  Ride lifecycle status enum
//

import Foundation

/// Represents the lifecycle status of a ride
enum RideStatus: String, Codable, CaseIterable {
    case requested = "requested"
    case searchingDriver = "searching_driver"
    case driverAssigned = "driver_assigned"
    case driverArriving = "driver_arriving"
    case inProgress = "in_progress"
    case completed = "completed"
    case cancelled = "cancelled"
    
    /// Human-readable display text
    var displayText: String {
        switch self {
        case .requested: return "Solicitando viaje..."
        case .searchingDriver: return "Buscando conductor..."
        case .driverAssigned: return "¡Conductor asignado!"
        case .driverArriving: return "El conductor está en camino"
        case .inProgress: return "Viaje en progreso"
        case .completed: return "Viaje completado"
        case .cancelled: return "Viaje cancelado"
        }
    }
    
    /// SF Symbol icon for the status
    var iconName: String {
        switch self {
        case .requested: return "clock.fill"
        case .searchingDriver: return "magnifyingglass"
        case .driverAssigned: return "checkmark.circle.fill"
        case .driverArriving: return "bicycle"
        case .inProgress: return "figure.outdoor.cycle"
        case .completed: return "flag.checkered"
        case .cancelled: return "xmark.circle.fill"
        }
    }
    
    /// Whether the ride is still active (not finished)
    var isActive: Bool {
        switch self {
        case .requested, .searchingDriver, .driverAssigned, .driverArriving, .inProgress:
            return true
        case .completed, .cancelled:
            return false
        }
    }
    
    /// Next status in the progression (for simulation)
    var nextStatus: RideStatus? {
        switch self {
        case .requested: return .searchingDriver
        case .searchingDriver: return .driverAssigned
        case .driverAssigned: return .driverArriving
        case .driverArriving: return .inProgress
        case .inProgress: return .completed
        case .completed, .cancelled: return nil
        }
    }
}
